featurebyte/routes/observation_table/controller.py

```diff
@@ -98,7 +101,16 @@ async def upload_observation_table_csv(
         -------
         Task
         """
-        observation_set_dataframe = dataframe_from_arrow_stream(observation_set_file.file)
+        # Check that filename is a .csv or .parquet file
+        uploaded_filename_extension = os.path.splitext(data.name)[-1]
+
+        if uploaded_filename_extension == ".csv":
+            observation_set_dataframe = dataframe_from_arrow_stream(observation_set_file.file)
+        elif uploaded_filename_extension == ".parquet":
+            observation_set_dataframe = pd.read_parquet(observation_set_file.file)
```

Comment:
we'll look to simplify this in the future if there's a performance issue with loading files into memory. if it becomes a problem, we can look to stream the parquet file all the way through to the warehouse connectors. will keep as is for now to keep it simple.